### ISSUE TITLE
spanner/spansql: Database option `version_retention_period` is not parsed correctly if there are tables after `ALTER DATABASE` statement

### ISSUE THREAD

**Issue by gecko655** [googleapis/google-cloud-go#5970]
**Client**

Cloud Spanner

**Environment**

Go playground

**Go Environment**

(This is the environment of my PC, not Go playground)

```
❯ go version
go version go1.17.5 darwin/arm64

❯ go env
GO111MODULE=""
GOARCH="arm64"
GOBIN=""
GOCACHE="/Users/[UserName]/Library/Caches/go-build"
GOENV="/Users/[UserName]/Library/Application Support/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="arm64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/[UserName]/.asdf/installs/golang/1.17.5/packages/pkg/mod"
GOOS="darwin"
GOPATH="/Users/[UserName]/.asdf/installs/golang/1.17.5/packages"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/Users/[UserName]/.asdf/installs/golang/1.17.5/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/Users/[UserName]/.asdf/installs/golang/1.17.5/go/pkg/tool/darwin_arm64"
GOVCS=""
GOVERSION="go1.17.5"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/[UserName]/git/google-cloud-go/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/_c/7hkzyhtj5mxc38jm7661stc00000gq/T/go-build755299587=/tmp/go-build -gno-record-gcc-switches -fno-common"

```

**Code**

```go
package main

import (
	"fmt"

	"cloud.google.com/go/spanner/spansql"
	"github.com/k0kubun/pp/v3"
)

func main() {
	pp.ColoringEnabled = false
	ddl, _ := spansql.ParseDDL("test", "ALTER DATABASE `test` SET OPTIONS (optimizer_version=4, version_retention_period = '7d', enable_key_visualizer=true);\n"+
		"\n"+
		"CREATE TABLE `users` (\n"+
		"    `UserId` STRING(36) NOT NULL,\n"+
		") PRIMARY KEY (`UserId`);")
	pp.Println(ddl)
	fmt.Println(ddl.List[0].SQL())
}

```
https://go.dev/play/p/I5rWmNCRvda

**Expected behavior**

`version_retention_period` should be parsed to `7d`

**Actual behavior**

`version_retention_period` is actually parsed to `UserId` (See more details: https://go.dev/play/p/I5rWmNCRvda )

Parsed DDL is "ALTER DATABASE test SET OPTIONS (optimizer_version=4, version_retention_period='UserId', enable_key_visualizer=true)"

**Additional context**

This behavior is not observed if there is no table after "ALTER DATABASE" statement.
In short, the following query is parsed correctly.

```go
spansql.ParseDDL("test", "ALTER DATABASE `test` SET OPTIONS (optimizer_version=4, version_retention_period = '7d', enable_key_visualizer=true);
```
---

Suspecting pointer passing implementation around here:
https://github.com/googleapis/google-cloud-go/blob/dd8973b7f72e4b153f7f135e4af8e89c66d3666d/spanner/spansql/parser.go#L1746
